x-pack/plugins/lens/public/app_plugin/show_underlying_data.ts

```diff
@@ -99,8 +99,15 @@ export function getLayerMetaInfo(
   const isVisible = Boolean(capabilities.navLinks?.discover && capabilities.discover?.show);
   // If Multiple tables, return
   // If there are time shifts, return
+  // If dataViews have not loaded yet, return
   const datatables = Object.values(activeData || {});
-  if (!datatables.length || !currentDatasource || !datasourceState || !activeVisualization) {
+  if (
+    !datatables.length ||
+    !currentDatasource ||
+    !datasourceState ||
+    !activeVisualization ||
+    !Object.keys(indexPatterns).length
```

Comment:
This is a bug fix which appears when a visualization started from a shared URL is saved, then the Lens editor would reload to change the URL using the SO id.
I suspect there's a race condition as `activeData` is available at that point but the dataViews cache has not been populated yet, but the editor is trying to load the `Explore in Discover` feature, and fails because of the lack of referenced dataView.

For now I've stopped the loading of the feature via this check, but perhaps it would be worth to either stop the editor reloading on SO saving or track down the race condition.